ggml.c

```diff
@@ -12490,12 +12551,24 @@ UseGgmlGemm1:;
             assert(params->wsize >= ne11*ne12*ne13*row_size);
             GGML_ASSERT(src1->type == GGML_TYPE_F32);
 
-            for (int64_t i13 = 0; i13 < ne13; ++i13) {
-                for (int64_t i12 = 0; i12 < ne12; ++i12) {
-                    for (int64_t i11 = 0; i11 < ne11; ++i11) {
-                        from_float_to_vec_dot((float *)((char *) src1->data + i13*nb13 + i12*nb12 + i11*nb11), (void *) wdata, ne10);
+            if (from_float_to_mat && gemm && (ne11 >= 4) && (ne12 == 1) && (ne13 == 1)) {
```

Comment:
The case of `(ne12 != 1) || (ne13 != 1)`, including 3D/4D matrix multiplications, will currently always fallback to the reference vec_dot `from_float_to_vec_dot` path. As a result, it will not be computed incorrectly, and we have not yet included any assert there. If you prefer, we can certainly include an assert there.